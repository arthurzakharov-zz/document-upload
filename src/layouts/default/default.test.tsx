import { render, screen } from "@testing-library/react";
import DefaultLayout from "./default.layout";

function Header() {
  return <span>Header</span>;
}
function Main() {
  return <span>Main</span>;
}
function Footer() {
  return <span>Footer</span>;
}

const HEADER = "default-layout-header";
const MAIN = "default-layout-main";
const FOOTER = "default-layout-footer";

describe("DefaultLayout", () => {
  test("Snapshot", () => {
    const { container } = render(<DefaultLayout header={Header} main={Main} footer={Footer} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("Each layout part is placed on correct place", () => {
    render(<DefaultLayout header={Header} main={Main} footer={Footer} />);
    expect(screen.getByTestId(HEADER)).toHaveTextContent("Header");
    expect(screen.getByTestId(MAIN)).toHaveTextContent("Main");
    expect(screen.getByTestId(FOOTER)).toHaveTextContent("Footer");
  });
});
