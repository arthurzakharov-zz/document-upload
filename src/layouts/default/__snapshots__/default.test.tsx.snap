// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout Snapshot 1`] = `
<div
  class="default-layout"
>
  <div
    class="default-layout__content"
  >
    <header
      data-testid="default-layout-header"
    >
      <span>
        Header
      </span>
    </header>
    <main
      data-testid="default-layout-main"
    >
      <span>
        Main
      </span>
    </main>
    <footer
      data-testid="default-layout-footer"
    >
      <span>
        Footer
      </span>
    </footer>
  </div>
</div>
`;
